parse/lex_test.go

```diff
@@ -300,6 +300,26 @@ func Test_lex(t *testing.T) {
 				mk(itemQuotedPatternClose, "}}"),
 			},
 		},
+		{
+			name:  "variable with _",
+			input: "{$csv_filename}",
+			expected: []item{
+				mk(itemExpressionOpen, "{"),
+				mk(itemVariable, "csv_filename"),
+				mk(itemExpressionClose, "}"),
+			},
+		},
+		{
+			name:  "variable with whitespace",
+			input: "{ $csv_filename }",
+			expected: []item{
+				mk(itemExpressionOpen, "{"),
+				mk(itemWhitespace, " "),
+				mk(itemVariable, "csv_filename"),
+				mk(itemWhitespace, " "),
+				mk(itemExpressionClose, "}"),
+			},
+		},
```

Comment:
In the given scenario, it's better to extend the existing test case "variable" - `"{ $max_users :math:round }"`. Otherwise, the number of tests will be unlimited.

Property testing would be the better approach to test name characters. Not part of this ticket.